src/lib/analytics/analytics.js

```diff
@@ -361,14 +365,31 @@ const patchLogger = () => {
   const debounceFireEvent = debounce(fireEvent, 500, { leading: true })
 
   logger.error = (...args) => {
-    const [logContext, logMessage, eMsg, errorObj, ...rest] = args
+    const [logContext, logMessage, eMsg, errorObj, _extra = {}] = args
+    let { dialogShown = true, category = ERROR_CATEGORY_UNEXPECTED, ...extra } = _extra
+    let errorToPassIntoLog = errorObj
+
+    if (
+      category === ERROR_CATEGORY_UNEXPECTED &&
+      ['connection', 'webSocket', 'network'].filter(str => eMsg.includes(str))
```

Comment:
```suggestion
      ['connection', 'webSocket', 'network'].some(str => eMsg.includes(str))
```

Reply:
will always set network category
if all tests failed, filter will return empty object
but **empty object isn't equal false but true**

there's `.some()` method performing actually you need